biocrnpyler/dna_construct.py

```diff
@@ -20,13 +20,284 @@
 from .dna_part_promoter import Promoter
 from .dna_part_rbs import RBS
 from .dna_part_terminator import Terminator
+import biocrnpyler.component_enumerator as ce
+from .component_enumerator import ComponentEnumerator,LocalComponentEnumerator, GlobalComponentEnumerator
 from .species import (ComplexSpecies, OrderedMonomer, OrderedPolymer,
                       OrderedPolymerSpecies)
 from .utils import all_comb, remove_bindloc, rev_dir
-
+import logging
 #integrase_sites = ["attB","attP","attL","attR","FLP","CRE"]
 
 
+class TxTlExplorer_CE(LocalComponentEnumerator):
+    def __init__(self,name="TxTlExplorer",possible_rxns=("transcription","translation"),\
+                    direction="forward",possible_directions=("forward","reverse"),\
+                        return_rnas = True, return_proteins = True):
+        """this class goes through a parts_list of a DNA_construct and decides what RNAs are made
+        and what proteins are made based on orientation and location of parts"""
+        LocalComponentEnumerator.__init__(self=self,name=name)
+        self.return_rnas = return_rnas
+        self.return_proteins = return_proteins
+        self.current_rnas = {}
+        self.current_proteins = {}
+        self.made_rnas = {}
+        self.all_rnas = {}
+        self.made_proteins = {}
+        self.current_rbs = None
+        self.possible_rxns = possible_rxns
+        self.direction=direction
+        self.second_looping = False
+        self.possible_directions = possible_directions
+    def enumerate_components(self,component):
+        #TODO the component we are given better be a dna_construct or something that can be txtled
```

Comment:
Great point. I think it is fair to enforce that this only works on DNAconstructs!